api.js

```diff
@@ -116,6 +116,19 @@ Api.prototype._run = function (files, _options) {
 	self.precompiler = new CachingPrecompiler(cacheDir, self.options.babelConfig);
 	self.fileCount = files.length;
 
+	var overwatch;
+	if (this.options.concurrency > 0) {
```

Comment:
I'm similarly worried about people using the `--concurrency` flag and seeing different behaviors based on their filename glob patterns. It's much simpler to convey in documentation "If the concurrency flag/option is present, automatic .only behavior is not supported" than a list of rules.

Additionally, a user manually running ava with a deliberate subset of tests is likely only doing so to run a few specific tests because they internally do not use `.only` (like my team) which actually cycles back up to my argument of nixing `.only` entirely since with the `--watch` flag and good file name globbing, there is no real need for `.only`.